main_app/views.py

```diff
@@ -1,13 +1,13 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render, redirect, get_object_or_404
```

Comment:
get_object_or_404 is a check for whether there is an item to .get() if there isn't, it returns error 404